pilot/cmd/istioctl/config.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"istio.io/istio/pkg/log"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+var (
+
+	// TODO - Add support for non-kube istio deployments
+	// TODO - Add config-diff to get the difference between what pilot wants and what the proxy has
+	// TODO - Add support for non-default proxy config locations
+	configCmd = &cobra.Command{
+		Use:   "proxy-config <podID>",
+		Short: "Retrieves proxy configuration for the specified pod [kube only]",
```

Comment:
Might want to clarify this currently only reports the static / bootstrap envoy config. Additional work is required to query the running envoy for LDS/RDS/CDS/SDS config.

Reply:
Agreed and clarified. Will add it to the TODO.